drivers/gpio/gpio.c

```diff
@@ -80,6 +80,16 @@ void gpio_set_value(int gpio, int value)
 	ops->set_value(gpio, value);
 }
 
+void gpio_set_pull(int gpio, int pull)
+{
+	assert(ops);
+	assert(ops->set_pull != 0);
+	assert((pull == PULLNONE) || (pull == PULLUP) || (pull == PULLDOWN));
+	assert(gpio >= 0);
+
+	ops->set_pull(gpio, pull);
```

Comment:
minor: Would be good to make it clear in the commit message that this is an optional function (since it is not asserted for in gpio_init()).